### ISSUE TITLE
Dictionary has random ordering

### ISSUE THREAD

**Issue by SuperUserNameMan** [godotengine/godot#1310]
Hi,

I'm using a recent windows custom build.

I don't know if it is a bug or a missing feature, but parsing a Dictionnary returns keys in random order.

```
func _ready():

    var dico = {
        A = "1 - April",
        B = "2 - Banana",
        C = "3 - Cocoa",
        D = "4 - Drum",
        E = "5 - Elephant",
    }

    for k in dico :
        print(k," : ",dico[k])
```

returns keys in this order :

```
C : 3 - Cocoa
D : 4 - Drum
E : 5 - Elephant
A : 1 - April
B : 2 - Banana
```

I fail to see the coherence of this behavior.
It could be far more usefull if the ordering was the same the keys were entered into the dictionnary, ~~or at least ordered alphabetically~~.

**Comment by philiplb** [comment]
A dictionary doesn't have an order by definition IMHO.

**Comment by randyyaj** [comment]
Yeah that functionality can be seen in many different languages. You can
sort the dictionary yourself.

**Comment by UsernameIsAReservedWord** [comment]
@philiplb :
I've just reread the wiki doc to remember where i got the certainty i could expect this behavior from Godot's dictionaries.
It is not said explicitly, but the comment into the "**GDScript/Language/For**" section, lets think Godot's dictionary is expected to iterate keys in a specific order (alphabetically or in the entered order). 

````
var dict = {"a":0, "b":1, "c":2}
for i in dict:
    print(dict[i])  # loop iterates the keys; with i being "a","b" and c" it prints 0, 1 and 2.
```` 
https://github.com/okamstudio/godot/wiki/gdscript#for

@randyyaj : thanks for this "alphabetical sorting" workaround. if you have a simple one for sorting them in the same order they were entered into the dictionary, you could save some lazy neurons of mines.

**Comment by randyyaj** [comment]
Yeah there is a way but it is not efficient at all.

Just store your keys into an array when you populate your dictionary.

    var dict = {"a":0, "b":1, "c":2}
    var dict_order = ["a","b","c"]
    for i in dict_order:
        print(dict[i])  # loop iterates the keys; with i being "a","b" and c" it prints 0, 1 and 2.

**Comment by UsernameIsAReservedWord** [comment]
@randyyaj : yep, that's what i've done. but since my dictionary contains a great number of keys, and that listing them manually into an extra array was error-prone, I simply created a `dict_push_back()` func that populated an extra key containing the list (that i inefficiently and lazily choose to stored as a comma separated string list).

```
func dico_push_back(dico, key, data):
	if dico.has("_key_order"):
		dico["_key_order"] += ","
	else:
		dico["_key_order"] = ""
	dico["_key_order"] += key
	
	dico[key] = data
````

**Comment by SuperUserNameMan** [comment]
ok, since random order is considered as an expected and globally accepted behavior for a dictionary, I'm closing this issue.

**Comment by bojidar-bg** [comment]
Probably fixed by 0a59c3c3a6a8c13a5c82d59d9587fca31f900604 though.
